executor/split.go

```diff
@@ -308,3 +313,162 @@ func (e *SplitTableRegionExec) getSplitTableKeys() ([][]byte, error) {
 	}
 	return keys, nil
 }
+
+// RegionMeta contains a region's peer detail
+type regionMeta struct {
+	region     *metapb.Region
+	leaderID   uint64
+	start      string
+	end        string
+	scattering bool
+}
+
+func getPhysicalTableRegions(physicalTableID int64, tableInfo *model.TableInfo, tikvStore tikv.Storage, s splitableStore) ([]regionMeta, error) {
+	uniqueRegionMap := make(map[uint64]struct{})
+	// for record
+	startKey, endKey := tablecodec.GetTableHandleKeyRange(physicalTableID)
+	regionCache := tikvStore.GetRegionCache()
+	recordRegionMetas, err := regionCache.LoadRegionsInKeyRange(tikv.NewBackoffer(context.Background(), 20000), startKey, endKey)
+	if err != nil {
+		return nil, err
+	}
+	recordRegions, err := getRegionMeta(recordRegionMetas, uniqueRegionMap)
+	if err != nil {
+		return nil, err
+	}
+
+	recordPrefix := tablecodec.GenTableRecordPrefix(physicalTableID)
+	tablePrefix := tablecodec.GenTablePrefix(physicalTableID)
+	decodeRegionsKey(recordRegions, tablePrefix, recordPrefix, nil, physicalTableID, 0)
```

Comment:
why don't move 
```go
	recordPrefix := tablecodec.GenTableRecordPrefix(physicalTableID)
	tablePrefix := tablecodec.GenTablePrefix(physicalTableID)
	decodeRegionsKey(recordRegions, tablePrefix, recordPrefix, nil, physicalTableID, 0)
```
into getRegionMeta, IMHO, decodeRegionsKey is also used to init the regionMeta.

Reply:
done.